### ISSUE TITLE
`admin-ajax.php` cache-exclusion regex could break on any non-root WordPress install (subdirectory install, or `WP_SITEURL` ≠ `WP_HOME`)

### ISSUE THREAD

**Issue by glouton** [wp-media/wp-rocket#8683]
**Describe the bug**
`CacheDynamicResource`'s exclusion pattern is built by joining `$excluded_files` with `|` and wrapping the result directly in `^...$` anchors, without a grouping construct around the alternation:

```php
// inc/Engine/Optimization/CacheDynamicResource.php
$excluded_files   = (array) apply_filters( 'rocket_exclude_static_dynamic_resources', [] );
$excluded_files[] = '/wp-admin/admin-ajax.php';
// ...
$this->excluded_files = implode( '|', $excluded_files );
// ...
if ( preg_match( '#^' . $this->excluded_files . '$#', $file['path'] ) ) {
```

Due to regex alternation having lower precedence than the anchors, `^` only binds to the *first* alternative and `$` only to the *last* one. This currently doesn't cause a visible problem on a default install: WP Rocket's own always-on `s2member` compatibility hook (`inc/3rd-party/plugins/s2member.php`, unconditionally registered regardless of whether s2member is installed) contributes a first entry to `$excluded_files`, and `/wp-admin/admin-ajax.php` is always appended last — so in practice its exclusion ends up effectively *right-anchored only* (`.../wp-admin/admin-ajax\.php$`, no `^`), which happens to still match on a subdirectory install or `WP_SITEURL` ≠ `WP_HOME` setup (WordPress's own "Giving WordPress its own directory" pattern).

This is incidental, not a deliberate design — it silently breaks the moment `$excluded_files` ends up with a single entry (e.g. if the `rocket_exclude_static_dynamic_resources` filter ever stops contributing anything, whether through a future refactor or another plugin removing/filtering that hook). At that point the pattern collapses to a single, fully-anchored `^/wp-admin/admin-ajax\.php$`, and `admin-ajax.php` silently stops being excluded on any non-root install.

**To Reproduce**
Verified directly via `wp eval` against `WP_Rocket\Engine\Optimization\CacheDynamicResource::is_excluded_file()` (WP Rocket 3.23.1):

```php
$options = new WP_Rocket\Admin\Options_Data( [] );
$cdr = new WP_Rocket\Engine\Optimization\CacheDynamicResource( $options, WP_ROCKET_CACHE_BUSTING_PATH, WP_ROCKET_CACHE_BUSTING_URL );

// Default state (s2member hook present) -- works, by accident:
var_dump( $cdr->is_excluded_file( 'https://example.com/wp/wp-admin/admin-ajax.php' ) ); // true

// After: remove_action( 'rocket_exclude_static_dynamic_resources', 'rocket_exclude_s2member_dynamic_files' );
// i.e. $excluded_files collapses to a single entry -- now broken:
var_dump( $cdr->is_excluded_file( 'https://example.com/wp/wp-admin/admin-ajax.php' ) ); // false, should be true
```

**Expected behavior**
`admin-ajax.php` should reliably be excluded regardless of how many other entries happen to be present in `$excluded_files`, and regardless of `WP_SITEURL`/install location.

**Screenshots**
Not applicable — server-side logic issue, not a rendering/UI bug.

**Server (please complete the following information):**
- WP Rocket: 3.23.1
- WordPress: 7.0.2
- PHP: 8.4.24
- Install type: Composer-managed (`roots/wordpress`), Bedrock-style directory layout, `WP_SITEURL` = `WP_HOME . '/wp'`

**Additional context**
Suggested fix: wrap the alternation in a non-capturing group before anchoring, `'#^(?:' . $this->excluded_files . ')$#'`, so the exclusion is correct regardless of how many entries end up in the list or their order — rather than relying on `admin-ajax.php` always being appended last after at least one other entry.

Found via code review while investigating unrelated `.htaccess`/vhost rewrite work, not from an observed production symptom. A search of this repo's issues turned up several past reports about path-handling under non-standard directory layouts (`#5603`, `#5534`, `#865`) — this seems to be a recurring pain point rather than a one-off, even though this particular instance is currently masked in default installs.
